### DISCUSSION TITLE
Migrating from 4.10 to 5.x and handling -Skip

### DISCUSSION THREAD

**Discussion by ChrisLynchHPE** [pester/Pester#2284] [Q&A] [answered]
I have a unit test script that looks like I can convert mostly to the new `Discovery` and `Run` format introduced in 5.x.  However, I have some conditional `-Skip` cases I'm not sure how to handle.  The PowerShell library I maintain interacts with a REST API.  Depending on certain behaviors, the `-Skip:$ShouldSkip` parameter needs to be used.  Now, this isn't put within a `BeforeAll` block, but rather in a previous `It` block.  I will set a variable that would skip the next unit test if the one generates a specific condition.  This is to handle defects with the API that either are found or need to be handled differently.

What is the best method to handle this case?

**Comment by fflaten** [comment, marked as answer]
Hi, that's a good question!

`-Skip` is evaluated as part of Discovery (see https://pester.dev/docs/usage/skip#conditional-skip). Ideally you'd run some code in `BeforeDiscovery` to figure out if you need them or not.

If that's not possible then you'd have to use `Set-ItResult` to skip. You will have to use a script-scoped (or worst case global) variable to share the variable between `It`-blocks.

`$script:`-scope is normally per `*.tests.ps1`-file unless you run code inside a module function or something like `InModuleScope SomeModule { .. }`. Remember to use unique variable names and/or clear it when it shouldn't apply anymore, ex. in a `AfterAll`. Especially with global-variable that would affect the next run.

```ps1
Describe 'Skipcheck in Run' {
    It 'first' {
        # Run logic to make decision.. To skip or not to skip
        $script:ShouldSkipInRun = $true
        1 | Should -Be 1
    }

    It 'second' {
        if ($script:ShouldSkipInRun) {
            Set-ItResult -Skipped -Because 'first test said so'
        }

        1 | Should -Be 1
    }
}
```

**Reply by ChrisLynchHPE** [reply]
Ah, that should work very well.  I have no issues using Global or Script scoped variables, as I use them within my existing unit tests.  It's just a few additional lines of code you provided, so that shouldn't be a big problem.
